src/main/java/es/unizar/smartcampuz/application/controller/AdminDashboardController.java

```diff
@@ -153,68 +159,65 @@
+    public ResponseEntity<String> approveOrDenyReservation (@RequestParam(name="id") long reservationId,
+                                                            @RequestParam(name="approved") boolean approved) throws IOException{
 
-        JSONObject json;
+        // TODO: Comprobar que findAllByRoomIDAndDateAndState() busca bien
+        Reservation reservation = reservationRepository.findOne(reservationId);
 
-        try{
-            json = JsonService.readJson(request.getReader());
-            reservationId = json.getInt("id");
-            approved = json.getBoolean("approved");
-        }
-        catch (Exception e){
-            return new ResponseEntity<>("\"Error interno en el servidor.\"", HttpStatus.INTERNAL_SERVER_ERROR);
+        if(reservation==null){
+            return new ResponseEntity<>("\"Reserva no encontrada.\"", HttpStatus.NOT_FOUND);
         }
 
-        // TODO: Pedir la reserva a la BD.
+        Set<Reservation> approvedReservations = reservationRepository.findAllByRoomIDAndDateAndState(reservation.getRoomID(),
+            reservation.getDate(), ReservationState.APPROVED);
+        List<Long> deniedReservations = new ArrayList<>();
 
-        ArrayList<Integer> removedReservations = new ArrayList<>();
+        //Checks compatibility policy with all given approved reservations
         JSONObject response = new JSONObject();
-        // TODO: Cambiar los parametros de la llamada del checkSchedule
-        if(approved && ReservationChecker.checkSchedule(new boolean[24], new ArrayList())){
-            // TODO: Asignar el estado "Approved" a la reserva
-
-            //TODO: Pedir a la BD reservas de la misma localización y día que la reserva aprobada
-            Iterable<?> iter = new ArrayList();
-            // TODO: Cogeré este array de la reserva que se ha aprobado
-            boolean [] reservedHours = new boolean[24];
-            boolean conflic = false;
-            for(Object o:iter){
-                // TODO: Cogeré este array del objeto Reservation
-                boolean [] reservation = new boolean[24];
-                for(int i = ReservationChecker.START_HOUR; i<= ReservationChecker.FINISH_HOUR && !conflic; i++){
-                    if(reservedHours[i] && reservation[i]){
-                        // TODO: Añadir el ID de la reserva que entra en conflicto
-                        removedReservations.add(1);
-                        conflic = true;
-                    }
+        if(approved && ReservationChecker.checkSchedule(reservation.getTimeReservation(), approvedReservations)){
+            //If it is compatible we save it as approved
+            reservation.setState(ReservationState.APPROVED);
+            reservationRepository.save(reservation);
+
+            //We look for the pending reservations that are hereinafter incompatible
+            Set<Reservation> pendingReservations = reservationRepository.findAllByRoomIDAndDateAndState(reservation.getRoomID(),
```

Comment:
I thought it was more readable in this way. It's also easier to debug and trace.